### PR TITLE
Add region setting & diff simulator

### PR DESCRIPTION
**Pull request by zhyupe**

### CODE REVIEW

**Comment by zhyupe** [comment]
@Supamiu I'm using your code of diff simulator and haven't dig into the control bouns problem yet

**Comment by Supamiu** [comment]
I think it's better to have the simulator entirely fixed before merging, but as always, it's a pretty impressive PR ^^

**Comment by Supamiu** [comment]
It might be worth to convert this commit's changes to your  new region approach: https://github.com/ffxiv-teamcraft/ffxiv-teamcraft/commit/03a435ab99fdff6c867097548cbe24a99bbb4239

**Comment by zhyupe** [comment]
> It might be worth to convert this commit's changes to your new region approach: [03a435a](https://github.com/ffxiv-teamcraft/ffxiv-teamcraft/commit/03a435ab99fdff6c867097548cbe24a99bbb4239)



> It might be worth to convert this commit's changes to your new region approach: [03a435a](https://github.com/ffxiv-teamcraft/ffxiv-teamcraft/commit/03a435ab99fdff6c867097548cbe24a99bbb4239)

Done :)

File: apps/client/src/app/core/simulation/simulation.service.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+export type CraftingJob = Simulator.CraftingJob;
+export type CrafterLevels = Simulator.CrafterLevels;
+export type CrafterStats = Simulator.CrafterStats;
+export type CraftingAction = Simulator.CraftingAction;
+export type EffectiveBuff = Simulator.EffectiveBuff;
+export type GearSet = Simulator.GearSet;
+export type Simulation = Simulator.Simulation;
+export type SimulationReliabilityReport = Simulator.SimulationReliabilityReport;
+export type SimulationResult = Simulator.SimulationResult;
+export type StepState = Simulator.StepState;
```
Comment (Supamiu) [inline apps/client/src/app/core/simulation/simulation.service.ts]: Are these still needed? I don't see them inside the other changed files.

Comment (zhyupe) [reply apps/client/src/app/core/simulation/simulation.service.ts]: https://github.com/ffxiv-teamcraft/ffxiv-teamcraft/pull/1373/files#diff-13914238e5098788f9e876d127449e86R59 Here are them imported

**Pull request merged**